src/main/java/com/jcabi/latex/maven/plugin/CompileMojo.java

```diff
@@ -99,6 +100,7 @@ public void execute() throws MojoFailureException {
         if (this.skip) {
             Logger.info(this, "execution skipped because of 'skip' option");
         } else {
+            this.checkOS();
```

Comment:
I don't think we need a separate method for this. It's a fairly simple check that we only do once, here.